lib/switchtec.c

```diff
@@ -395,7 +403,10 @@ void switchtec_status_free(struct switchtec_status *status, int ports)
  */
 const char *switchtec_strerror(void)
 {
-	const char *msg;
+	const char *msg = "Unknown error";
+
+	if (mrpc == -1)
+		return strerror(errno);
```

Comment:
I think you need to call platform_strerror() in this path as well...

Reply:
Ok, I found current Linux platform_strerror() simply returns "Success", I'll make it wrapping the strerror().